roles/bootstrap-os/tasks/bootstrap-centos.yml

```diff
@@ -53,12 +53,12 @@
     state: present
   when:
     - not is_atomic
-    - package_python_pip.results | length != 0
+    - package_python_pip.results | length == 0
```

Comment:
I think this is backwards.
1. The preceding task `Check python-pip package` should probably come after the epel install, since epel is what typically makes that package available.
2. The 'results' list from `yum: list=python-pip` will contain results when `python-pip` is installed *or* available. I believe the original intent of the `package_python_pip.results | length != 0` condition was to only attempt to install pip if it's in the list of results.

From CI, you can see that `yum: list=python-pip` returns an empty list before epel is installed:
```
TASK [bootstrap-os : Check python-pip package] *********************************
[1;30mtask path: /kargo-ci/kubernetes-sigs-kubespray/roles/bootstrap-os/tasks/bootstrap-centos.yml:43[0m
Wednesday 23 January 2019  09:02:18 +0000 (0:00:15.951)       0:00:34.293 ***** 
[0;32mok: [k8s-44379687-149573517-1] => {"changed": false, "results": []}[0m
[0;32mok: [k8s-44379687-149573517-2] => {"changed": false, "results": []}[0m
[0;32mok: [k8s-44379687-149573517-3] => {"changed": false, "results": []}[0m

TASK [bootstrap-os : Install epel-release for bootstrap] ***********************
[1;30mtask path: /kargo-ci/kubernetes-sigs-kubespray/roles/bootstrap-os/tasks/bootstrap-centos.yml:50[0m
Wednesday 23 January 2019  09:02:21 +0000 (0:00:02.895)       0:00:37.188 ***** 
[0;32mok: [k8s-44379687-149573517-3] => {"changed": false, "msg": "", "rc": 0, "results": ["epel-release-7-11.noarch providing epel-release is already installed"]}[0m
[0;32mok: [k8s-44379687-149573517-1] => {"changed": false, "msg": "", "rc": 0, "results": ["epel-release-7-11.noarch providing epel-release is already installed"]}[0m
[0;32mok: [k8s-44379687-149573517-2] => {"changed": false, "msg": "", "rc": 0, "results": ["epel-release-7-11.noarch providing epel-release is already installed"]}[0m
```

Reply:
if `yum: list=python-pip` gives an empty list, that means it's not installed, right? Then it makes sense to install the `epel-release` package to install Fedora repositories and install python-pip packages from it. But for people that already have python-pip installed on their system (by whatever means, for example not from Fedora repositories), you shouldn't have to install `epel-release` to proceed to next steps.

Reply:
My point is, for online installations that can't rely on `epel-release` package, we need to add the python-pip package as requirement and check if it's already installed, and fails otherwise. 
The thing is, python-pip is a meta package, it's provided by either python2-pip or python3-pip, so I'm not sure how we can address this

Reply:
We could use get-pip.py for a distribution independent method. https://pip.pypa.io/en/stable/installing/

Reply:
Does Python on CentOS not ship with `ensurepip` built in?

Also: Is `pip` even required to be installed during the bootstrap role? Where would it be used later on?

Reply:
and I don't like this warning much: 

> Warning
> 
> Be cautious if you are using a Python install that is managed by your operating system or another package manager. get-pip.py does not coordinate with those tools, and may leave your system in an inconsistent state.